packages/docusaurus-theme-classic/src/theme/DocSidebar/index.tsx

```diff
@@ -78,20 +85,30 @@ function DocSidebarItemCategory({
     if (justBecameActive && collapsed) {
       setCollapsed(false);
     }
-  }, [isActive, wasActive, collapsed]);
-
-  const handleItemClick = useCallback(
-    (e) => {
-      e.preventDefault();
-
-      if (!menuListHeight) {
-        handleMenuListHeight();
+    if (goTo && collapsed) {
+      for (const i in items) {
+        if (items.hasOwnProperty(i)) {
+          const childItem = items[i];
+          if (childItem.type === 'link') {
+            if (goTo.includes(childItem.label.toLowerCase())) {
+              setIntitalHref(childItem.href);
+              setGoToLink(true);
+            }
+          }
+        }
       }
+    }
+  }, [isActive, wasActive, collapsed]);
 
-      setTimeout(() => setCollapsed((state) => !state), 100);
-    },
-    [menuListHeight],
-  );
+  const handleItemClick = useCallback(async () => {
+    if (!menuListHeight) {
+      handleMenuListHeight();
+    }
+    setCollapsed((state) => {
+      return !state;
+    });
+    await sleep(100);
```

Comment:
Why do you need that sleep here?

Reply:
Before `setCollapsed` was inside a setTimeout, for animation purposes for the categories opening and closing I just removed setTimeout and replacing it with sleep which does the same thing and purpose.